### ISSUE TITLE
Error memo_wising in subclass with included module

### ISSUE THREAD

**Issue by tindron** [panorama-ed/memo_wise#239]
```ruby
require 'memo_wise'

class C1
  prepend MemoWise
  def method_one
    1
  end
  memo_wise :method_one
end

module M1
  prepend MemoWise
  def method_two
    2
  end
  memo_wise :method_two
end

class C2 < C1
  include M1
  def method_three
    3
  end
  memo_wise :method_three
end
```
Results in:
```
/Users/randy.stoller/.rvm/gems/ruby-3.0.2/gems/memo_wise-1.3.0/lib/memo_wise/internal_api.rb:219:in `class_variable_get': class variable @@_memo_wise_index_counter of M1 is overtaken by C1 (RuntimeError)
	from /Users/randy.stoller/.rvm/gems/ruby-3.0.2/gems/memo_wise-1.3.0/lib/memo_wise/internal_api.rb:219:in `next_index!'
	from /Users/randy.stoller/.rvm/gems/ruby-3.0.2/gems/memo_wise-1.3.0/lib/memo_wise.rb:181:in `memo_wise'
	from test.rb:24:in `<class:C2>'
	from test.rb:19:in `<main>'
```
This worked in 1.1.0 and only happens if the superclass, subclass and module all `memo_wise` something.

**Comment by JacobEvelyn** [comment]
Thanks for reporting this, @tindron ! This looks similar to #238; @jemmaissroff and I will work on a solution that fixes both issues.

**Comment by JacobEvelyn** [comment]
@tindron we just released v1.4.0, which fixes this issue. Thanks so much for your patience. ❤️

Please try it out and let us know if you encounter any other issues!
